lib/grape/middleware/stack.rb

```diff
@@ -6,14 +6,27 @@ module Middleware
     # It allows to insert and insert after
     class Stack
       class Middleware
-        attr_reader :args, :block, :klass
+        use_ruby2_keywords = Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('2.7')
 
-        def initialize(klass, *args, &block)
+        attr_reader :args, :opts, :block, :klass
+
+        def initialize(klass, *args, **opts, &block)
           @klass = klass
-          @args = args
+          @args  = args
+          @opts  = opts
           @block = block
         end
 
+        def use_in(builder)
```

Comment:
Should `use_in` be private?

Reply:
makes sense, will update

Reply:
actually, no, it's called from `Stack#build`